meta/downloadHelperBinaries.ts

```diff
@@ -0,0 +1,193 @@
+import { createWriteStream } from 'fs'
+import { chmod, stat, mkdir, readFile, writeFile } from 'fs/promises'
+import { dirname, join } from 'path'
+import { Readable } from 'stream'
+import { finished } from 'stream/promises'
+
+type SupportedPlatform = 'win32' | 'darwin' | 'linux'
+type DownloadedBinary = 'legendary' | 'gogdl' | 'nile'
+
+const RELEASE_TAGS = {
+  legendary: '0.20.35',
+  gogdl: 'v1.1.1',
+  nile: 'v1.1.0'
+} as const satisfies Record<DownloadedBinary, string>
+
+const pathExists = async (path: string): Promise<boolean> =>
+  stat(path).then(
+    () => true,
+    () => false
+  )
+
+async function downloadFile(url: string, dst: string) {
+  const response = await fetch(url, {
+    headers: {
+      'User-Agent': 'HeroicBinaryUpdater/1.0'
+    }
+  })
+  await mkdir(dirname(dst), { recursive: true })
+  const fileStream = createWriteStream(dst, { flags: 'w' })
+  await finished(Readable.fromWeb(response.body).pipe(fileStream))
+}
+
+async function downloadAsset(
+  binaryName: string,
+  repo: string,
+  tag_name: string,
+  arch: string,
+  platform: SupportedPlatform,
+  filename: string
+) {
+  const url = `https://github.com/${repo}/releases/download/${tag_name}/${filename}`
+  console.log('Downloading', binaryName, 'for', platform, arch, 'from', url)
+
+  const exeFilename = binaryName + (platform === 'win32' ? '.exe' : '')
+  const exePath = join('public', 'bin', arch, platform, exeFilename)
+  await downloadFile(url, exePath)
+
+  console.log('Done downloading', binaryName, 'for', platform, arch)
+
+  if (platform !== 'win32') {
+    await chmod(exePath, '755')
+  }
+}
+
+/**
+ * Downloads assets uploaded to a GitHub release
+ * @param binaryName The binary which was built & uploaded. Also used to get the final folder path
+ * @param repo The repo to download from
+ * @param tag_name The GitHub Release tag which produced the binaries
+ * @param assetNames The name(s) of the assets which were uploaded, mapped to platforms
+ */
+async function downloadGithubAssets(
+  binaryName: string,
+  repo: string,
+  tag_name: string,
```

Comment:
```suggestion
  tagName: string,
```

for consistency with the case of the other arguments